api/v1alpha1/extension_types.go

```diff
@@ -82,6 +82,11 @@ type ExtensionSpec struct {
 
 	// source of Extension to be installed
 	Source ExtensionSource `json:"source"`
+
+	//+kubebuilder:Optional
+	//
+	// skipCRDUpgradeSafetyCheck specifies whether or not the CRD upgrade safety checks should be skipped when attempting to install the extension
+	SkipCRDUpgradeSafetyCheck bool `json:"skipCRDUpgradeSafetyCheck,omitempty"`
```

Comment:
I think in the past we identified that bool fields are problematic over time since quite frequently an additional value would slip in, so for maximum flexibility we've been using enum/string for this role, like [so](https://github.com/operator-framework/operator-controller/blob/main/config/crd/bases/olm.operatorframework.io_clusterextensions.yaml#L46-L52). 
Is there any reason not to do so here?

Reply:
I don't disagree, but I recall us having also gone back and forth on our feelings on that. I'm happy to make a change to this, but I do think we should make sure we come up with a consistent approach as a project and that we all agree to prevent flip-flopping.

Reply:
IMO, let's go ahead and get this merged to unblock further development. We'll separately plan for a full API review once we have all of the MVP functionality complete, and do any necessary refactoring at that point.